bundle/config/mutator/process_target_mode.go

```diff
@@ -130,8 +130,17 @@ func validateProductionMode(ctx context.Context, b *bundle.Bundle, isPrincipalUs
 		}
 	}
 
-	if !isPrincipalUsed && !isRunAsSet(r) {
-		return diag.Errorf("'run_as' must be set for all jobs when using 'mode: production'")
+	// We need to verify that there is only a single deployment of the current target.
+	// The best way to enforce this is to explicitly set root_path.
+	if !isExplicitRootSet(b) {
+		if isRunAsSet(r) || isPrincipalUsed {
+			// Just setting run_as is not enough to guarantee a single deployment,
+			// and neither is setting a principal.
+			// We only show a warning for these cases since we didn't historically
+			// report an error for them.
```

Comment:
So this is breaking IFF:
1. You're a regular user
2. You don't have `run_as` set (i.e. it runs as self)

?

Reply:
I do think it's a good practice to set `run_as` despite its limitations and it seems okay to set it in the templates. Providing a warning when you only set `run_as` and don't set `root_path` seems like a sweet spot to me. Providing an error would be a breaking change and doesn't really seem warranted.

Reply:
Do you think we should de-emphasize `run_as` in general?

Reply:
`run_as` still seems helpful to me and it will become much more usable once we add support for it in pipelines.